src/org/sosy_lab/common/collect/SkipListTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *  SoSy-Lab Common is a library of useful utilities.
+ *  This file is part of SoSy-Lab Common.
+ *
+ *  Copyright (C) 2007-2015  Dirk Beyer
+ *  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.sosy_lab.common.collect;
+
+import com.google.common.collect.testing.SortedSetTestSuiteBuilder;
+import com.google.common.collect.testing.TestStringSortedSetGenerator;
+import com.google.common.collect.testing.features.CollectionFeature;
+import com.google.common.collect.testing.features.CollectionSize;
+import com.google.common.testing.SerializableTester;
+import com.google.errorprone.annotations.Var;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.SortedSet;
+import junit.framework.JUnit4TestAdapter;
+import junit.framework.TestSuite;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SkipListTest {
+
+  private static class TestSkipListGenerator extends TestStringSortedSetGenerator {
+
+    @Override
+    protected SortedSet<String> create(String[] pIntegers) {
+      SkipList<String> list = new SkipList<>();
+      // noinspection ResultOfMethodCallIgnored
+      boolean changed = list.addAll(Arrays.asList(pIntegers));
+      assert list.isEmpty() || changed;
+
+      return list;
+    }
+  }
+
+  public static junit.framework.Test suite() {
+    TestSuite suite =
+        SortedSetTestSuiteBuilder.using(new TestSkipListGenerator())
+            .named("SkipList Test Suite")
+            .withFeatures(
+                CollectionSize.ANY,
+                CollectionFeature.SUPPORTS_ADD,
+                CollectionFeature.SUPPORTS_REMOVE,
+                CollectionFeature.SUPPORTS_ITERATOR_REMOVE,
```

Comment:
The above three can be replaced with `SetFeature.GENERAL_PURPOSE`.

Probably we also want `CollectionFeature.KNOWN_ORDER`, `CollectionFeature.ALLOWS_NULL_VALUES` (depending on what we decide), and `CollectionFeature.SERIALIZABLE_INCLUDING_VIEWS`.